src/SourceLocator/SourceLocator.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Asgrim\SourceLocator;
+
+interface SourceLocator
```

Comment:
Actually very good point. You're locating the source code that "should" contain the class. Similarly, a new interface should be created for functions. So propose:

```php
<?php
interface ClassSourceLocator {
  public function __invoke($className);
}
interface FunctionSourceLocator { // for future
  public function __invoke($functionName);
}

class ClassReflector { // was Reflector
  public function __construct(ClassSourceLocator $sourceLocator);
  public function reflect($className);
}
class FunctionReflector { // for future
  public function __construct(FunctionSourceLocator $sourceLocator);
  public function reflect($functionName);
}
```

Happy to discuss other options, just an idea :)